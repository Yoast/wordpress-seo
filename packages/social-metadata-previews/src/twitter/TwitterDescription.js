/* External dependencies */
import styled from "styled-components";
import { TWITTER_IMAGE_SIZES } from "../helpers/determineImageProperties";

/**
 * Renders a TwitterDescription component.
 * NOTE: the "-webkit-line-clamp: 3" limits the number of lines to 3.
 *
 * @param {object} props The props.
 *
 * @returns {React.Component} The rendered element.
 */
const TwitterDescription = styled.p`
	max-height: 55px;
	min-height: 20px;
	overflow: hidden;
	font-size: 15px;
	line-height: 20px;
	text-overflow: ellipsis;
	margin: 0 0 2px;
	color: rgb(101, 119, 134);
	display: -webkit-box;
	cursor: pointer;
	-webkit-line-clamp: 2;
	-webkit-box-orient: vertical;

	@media all and ( max-width: ${ TWITTER_IMAGE_SIZES.landscapeWidth }px ) {
		display: none;
	}
`;

export default TwitterDescription;
