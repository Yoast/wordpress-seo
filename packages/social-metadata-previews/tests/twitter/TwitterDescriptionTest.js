/* External dependencies */
import React from "react";
import renderer from "react-test-renderer";

/* Internal dependencies */
import TwitterDescription from "../../src/twitter/TwitterDescription";

describe( "TwitterDescription", () => {
	it( "matches the snapshot for the summary card", () => {
		const component = renderer.create(
			<TwitterDescription
				description={
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description."
				}
				isLarge={ false }
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "matches the snapshot for the summary with large image card", () => {
		const component = renderer.create(
			<TwitterDescription
				description={
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description. " +
					"A very long description. A very long description. A very long description. A very long description."
				}
				isLarge={ true }
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
