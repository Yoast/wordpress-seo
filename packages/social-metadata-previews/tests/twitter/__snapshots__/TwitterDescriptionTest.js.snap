// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitterDescription matches the snapshot for the summary card 1`] = `
.c0 {
  max-height: 55px;
  min-height: 20px;
  overflow: hidden;
  font-size: 15px;
  line-height: 20px;
  text-overflow: ellipsis;
  margin: 0 0 2px;
  color: rgb(101,119,134);
  display: -webkit-box;
  cursor: pointer;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

@media all and ( max-width:506px ) {
  .c0 {
    display: none;
  }
}

<p
  className="c0"
/>
`;

exports[`TwitterDescription matches the snapshot for the summary with large image card 1`] = `
.c0 {
  max-height: 55px;
  min-height: 20px;
  overflow: hidden;
  font-size: 15px;
  line-height: 20px;
  text-overflow: ellipsis;
  margin: 0 0 2px;
  color: rgb(101,119,134);
  display: -webkit-box;
  cursor: pointer;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}

@media all and ( max-width:506px ) {
  .c0 {
    display: none;
  }
}

<p
  className="c0"
/>
`;
